contracts/DepositLimitModule.vy

```diff
@@ -0,0 +1,123 @@
+# pragma version ~=0.4
+
+"""
+@title Deposit Limit Module for yearn vaults
+@notice This contract controls deposits into yearn vault, and additionally manages roles (admin and controller).
+@dev Admins can appoint new admins and controllers, while controllers can pause or resume deposits.
+"""
+
+
+################################################################
+#                           STORAGE                            #
+################################################################
+
+# Two main roles: admin and controller
+is_admin: public(HashMap[address, bool])
+is_controller: public(HashMap[address, bool])
+
+# Main contract state for deposit control
+deposits_paused: public(bool)
+
+
+################################################################
+#                         CONSTRUCTOR                          #
+################################################################
+
+@deploy
+def __init__():
+    """
+    @notice Initializes the contract by assigning the deployer as the initial admin and controller.
+    """
+    self._set_admin(msg.sender, True)
+    self._set_controller(msg.sender, True)
+    self._set_deposits_paused(False)
```

Comment:
redundant assignment here

Reply:
while it's false by default, i decide to be explicit here